src/minimum_tracer/minimum_tracer.cpp

```diff
@@ -0,0 +1,2903 @@
+            ConvertToNonFlatDirections(zeroTemp);
+            newMinimum.point     = zeroTemp;
+            newMinimum.temp      = currentT;
+            newMinimum.potential = V(zeroTemp) * (1 + currentT * currentT);
+
+            if (old_min_is_global) // check if newMinimum is
+                                   // still global minimum
+            {
+              IsGlobMin(newMinimum);
+              if (!newMinimum.is_glob_min)
+              {
+                if (output)
+                  Logger::Write(
+                      LoggingLevel::MinTracerDetailed,
+                      "Phase no longer coincides with global minimum at " +
+                          std::to_string(newMinimum.temp));
+                globMinEndT       = newMinimum.temp;
+                old_min_is_global = false;
+              }
+            }
+
+            MinimumList.push_back(newMinimum);
+          }
+          // End tracking
+          break;
+        }
+        else
+        {
+          ss << "Calculation of phase tracker failed. T  = " << currentT
+             << " GeV\t|\t Final T = " << finalT << " GeV\t|\t"
+             << LengthGradient << "\t|\t"
+             << SmallestEigenvalue(new_point, Hessian) << "\t";
+          // Sucess saddle point!
+          ss << "\033[1;31m.\033[0m";
+          if (output) Logger::Write(LoggingLevel::MinTracerDetailed, ss.str());
+          ss.str(std::string());
+          return MinimumList; // return if starts in a saddle point
+        }
+        IsInMin = 1;
+      }
+      else
+      {
+        if (IsInMin == 1)
+        {
+          // We should never be here. We only track minimum
+          break;
+        }
+        else
+        {
+          newMinimum.point     = new_point;
+          newMinimum.temp      = currentT;
+          newMinimum.potential = V(new_point) * (1 + currentT * currentT);
+
+          if (old_min_is_global) // check if newMinimum is
+                                 // still global minimum
+          {
+            IsGlobMin(newMinimum);
+            if (!newMinimum.is_glob_min)
+            {
+              if (output)
+                Logger::Write(
+                    LoggingLevel::MinTracerDetailed,
+                    "Phase no longer coincides with global minimum at " +
+                        std::to_string(newMinimum.temp));
+              globMinEndT       = newMinimum.temp;
+              old_min_is_global = false;
+            }
+          }
+
+          MinimumList.push_back(newMinimum);
+          // Sucess minimum!
+          ss << "\033[1;32m.\033[0m";
+          dT *= .5 * ThresholdDistance /
+                Distance; // Try to predict the best stepsize
+          if (abs(initialdT) <= abs(dT)) dT = initialdT;
+        }
+        IsInMin = -1;
+      }
+      point = new_point;
+    }
+
+    // Make sure that or step is not bigger than it should be and we overshot
+    // the next minimum
+    if (currentT == finalT) break;
```

Comment:
CodeQL warns here about float comparison. Do you want here an exact comparison or an almost-the-same comparison?

Reply:
Yes